src/main/java/org/sonarqube/auth/bitbucket/BitbucketIdentityProvider.java

```diff
@@ -133,6 +137,32 @@ private GsonEmails requestEmails(OAuthService scribe, Token accessToken) {
     return null;
   }
 
+  private void checkTeamRestriction(OAuthService scribe, Token accessToken, GsonUser user) {
+    String[] teamsRestriction = settings.teamRestriction();
+    if (teamsRestriction != null && teamsRestriction.length > 0) {
+      GsonTeams userTeams = requestTeams(scribe, accessToken);
+      if (userTeams == null || userTeams.getTeams() == null || userTeams.getTeams().isEmpty()) {
+        throw new UnauthorizedException(format("No teams found for user %s", user.getUsername()));
```

Comment:
I would be tempted to merge this use case with the next one : we would fail with the same error "User is not part of..." if the user has no team or doesn't belong to restricted team